MH-13307, Update Release Manager Documentation

lkiesow opened the pull request:
The release manager documentation was heavily outdated. This new version
tries to reflect the current responsibilities of release managers
instead, giving new candidates an idea about the work involved in this
role.

lkiesow committed:
MH-13307, Update Release Manager Documentation

The release manager documentation was heavily outdated. This new version
tries to reflect the current responsibilities of release managers
instead, giving new candidates an idea about the work involved in this
role.

lkiesow commented:
@mtneug, since you are one of the latest release managers, would you mind reading through the docs and check if anything is amiss?

gregorydlogan commented:
Assigning myself for copy-editing purposes.  Will look later today.

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-Role
-----
+The single most important duty of release managers is to keep an eye on their release, timely notify the community about
```
timely notify the community about possible problems -> notify the community about possible problems in a timely manner

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-The release manager is ultimately responsible for making sure that the development process is being followed during his
-or her release.
+Expectations of the community towards releases from which problems can arise and which should be handles be release
```
The community has a number of expectations for release managers, and their handling of the problems which may arise during the release cycle.  The core of these expectations are:

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-* To ensure that the release is timely and does not languish in process.
-* To encourage committers to be involved in testing and fixing of bugs.
+This means that a release manager may need to force decisions around the release, help negotiate the acceptance or
+rejection of contributions and provide regular updates about the release on list, during the technical and the adopter
```
list, and during the technical and adopter

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-manager. The committer list must then either vote within 72 hours on a new release manager, or consider the release
-abandoned (the major, minor, and maintenance numbers remain unchanged).
+While a general rule is certainly just to look out for the release, work together with the community to make the release
+work properly and be pragmatic about the process, there are a few tasks a release manager has to ensure that they are
```
there are a few tasks which really can only be done by the release manager

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-------
+For all of these tasks, it's generally a good idea to look at previous releases and at their solutions for the tasks.
+Often, completing the task is simply a matter of repeating or updating previous work (e.g. advance the previous release
+schedule by six month).
```
six months

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-### Setting Release Schedule
+### Release Notes
+
+It's usually a good idea to create or clean the release notes page. This allows for a place to put the release schedule
```
clean the release notes page early in the release cycle.

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 Releases should happen twice a year, usually within a time span of 9.5 months between the cut of the previous release
-branch and the final release. The release manager should create a release schedule as soon as possible.
-For more details about the release schedule, have a look at the [Development Process](development-process.md)
+branch and the final release. The release managers should create a release schedule as soon as possible, announce it on
```
Re manager vs managers, we're using manager in most other places so I would say we either change *all* of them on this page, or none of them.

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-branch is created off of `develop` and should be named `r/N.x` (e.g. `r/3.x` for the Opencast 3.0 release branch).
-In the following command, we assume that the release branch for 3.0 is to be created. For other versions, please adjust
-the version number accordingly.
+for a given release.  This branch is split of `develop` and should be named `r/N.x` (e.g. `r/6.x` for the Opencast 6
```
is split off `develop`

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-release. You may be able to do this yourself by assigning a new fix version to their newly created merge ticket. If you
-are unable to do that, please contact one of the JIRA administrators so we can create it for you!
+To save the <LANGUAGE> translation from removal, it needs to
+be translated at least 90% until <DATE>.
```
until -> before

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+        git merge <remote>/develop   # this should be a fast-forward merge
 
-4. Merge the release branch. Not that if large merge conflicts arise, you may ask for help from the people creating the
+3. Merge the release branch. Not that if large merge conflicts arise, you may ask for help from the people creating the
```
Note that if

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-If the version is a release candidate, you probably want to highlight that there are no *Blockers* left at the moment
-and *#propose* this to become the final release.
+Updating the [localization translations](localization.md) is automated for existing translation files. If new files need
+to be added, it is is something that should happen early during the release process.
```
it ~~is~~ is

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-        git merge-base 3.0-rc2 r/3.x
-          060dfc3e2328518ae402846577fc6e7ce3b650d7
+        git checkot r/6.x
```
checkot -> checkout

gregorydlogan commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-    manually.
+        git push <remote> 6.0:6.0
 
-14. Update [the website](https://github.com/opencast/opencast.github.io) with a pull request, or directly if you have
```
Is there a reason you removed this section?  This gets missed *all* the time as is, if we don't remind RMs in the docs then it won't ever happen.

lkiesow replied:
I was surprised to find it in here at all and I wouldn't see that as part of the responsibility of a release manager to manage the website. The website is the board's responsibility from my point of view. Release managers should not have to deal with that like they also don't need to feed Twitter or Facebook.

In general, I would like to not push a lot of additional responsibilities towards release managers or else, over time it will only become harder to find volunteers for the job.

But that's something we can certainly discuss in the community. I'll bring that to list.

gregorydlogan replied:
Per the discussion in the technical meeting, I withdraw my objection here.  We'll figure out a way to automate this.

gregorydlogan reviewed (changes requested):
Looks good other than the changes identified.

lkiesow committed:
MH-13307, Release Documentation Fixes

This patch adds a few minor fixes (typos, wording) to the release
manager documentation.

lkiesow commented:
Fixed everything except the website which I will bring to list.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+
+It's usually a good idea to create or clean the release notes page. This allows for a place to put the release schedule
+and makes it easy to encourage people to put short descriptions of features there while these features are reviewed.
+This makes it easier to later create the final release notes.
```
Is it encouraged to update the release notes in each feature pull request? This will lead to many merge conflicts. Maybe it is better to include a usable description in the pull request and use the feature tag. Before the release, it is then a matter of going through the feature pull requests and copy-paste the description.

lkiesow replied:
I wouldn't expect many conflicts. Where would you see them? between different features going in `develop`? Would you just drop this whole paragraph?

mtneug replied:
Take for instance two feature pull requests A and B. If both update the release notes with their description, the release notes file will create a merge conflict if either pull request is merged. This gets worse with more simultaneous pull requests. Additionally, if a feature does not make it into `develop` before the feature freeze, the author would have to update the pull requests after the release notes file has been cleaned again.

lkiesow replied:
Again, I wouldn't expect that to be a problem with the current number of features but fair enough, I removed the recommendation. Let release manager decide that on their own if they want to.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+        git checkout -b r/7.x
+        git push <remote> r/7.x
+
+5. That is it for the release branch. Now update the versions in `develop` in preparation for the next release:
```
Not that it matters in Markdown, but this should be `4.` (continued below).

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
         git checkout develop
         for i in `find . -name pom.xml`; do \
-          sed -i 's/<version>3.0-SNAPSHOT</<version>4.0-SNAPSHOT</' $i; done
+          sed -i 's/<version>6-SNAPSHOT</<version>7-SNAPSHOT</' $i; done
```
Isn't this example for Opencast 7.x? Then we need to update `develop` to `8-SNAPSHOT`.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-        git commit -as
+        git add $(git status | grep 'modified:.*pom.xml' | awk '{print $2;}')
+        git commit -s
```
Maybe suggesting the message with `-m "NOJIRA: Bumping version numbers to 8-SNAPSHOT"` would be helpful?

lkiesow replied:
I personally have a strong aversion against the `NOJIRA` since it gives you no additional information at all. Others seem to like it though.

To not argue about this, I suggest we avoid the messages for now. Right now, the only place we have a suggestion is the release commit which does not use the `NOJIRA` and [in practice mostly hasn't as well](https://paste.fedoraproject.org/paste/XziB7-s9BBXbdUHNSjGgrQ).

mtneug replied:
I don't care either way; `NOJIRA` can be omitted. I just thought that having a common predefined message would simplify things.

lkiesow replied:
Okay, I've added some defaults

mtneug commented on docs/guides/developer/docs/release-manager.md:
````diff
@@ -1,474 +1,366 @@
+As always, we hope to have a lot of people testing this
+version, especially during the public QA phase.  Please
+report any bugs or issues you encounter.
+```
````
Maybe it would be a good idea to advertise the test server? I also build daily Docker images for the upcoming release (right know form `develop` after the cut from the release branch until release). @lkiesow you have done so with the RMP packages for 6.x if I'm not mistaken.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-    Hi everyone,
+When the release branch is cut, the release manager should check if there are languages to be in- or excluded for the
+upcoming release as specified [by the criteria in the localization documentation](localization.md).  If any modification
+criteria are met, an announcement should be published on the Opencast users list that specifies:
```
Do I understand this sentence correctly that no mail should be send in case the list of supported translations does not change? Shouldn't the "included translation mail" be always send?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+[1] Opencast project on Crowdin
+    https://crowdin.com/project/opencast-community
+[2] Inclusion and Exclusion of Translations
+    https://docs.opencast.org/develop/developer/...
```
Use <https://docs.opencast.org/develop/developer/localization/#inclusion-and-exclusion-of-translations> as link?

mtneug commented on docs/guides/developer/docs/release-manager.md:
````diff
@@ -1,474 +1,366 @@
+```
 
-### Creating the Merge Ticket
+Example announcement for endangered languages. Please create a post for each endangered language:
````
If there are many endangered languages, a single mail could be send?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-the [administrator](https://docs.opencast.org/develop/developer/infrastructure/#administrators) of that
-tool to ensure the ticket is added.
+[1] Inclusion and Exclusion of Translations
+    https://docs.opencast.org/
```
Use <https://docs.opencast.org/develop/developer/localization/#inclusion-and-exclusion-of-translations> as link?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+### Merging Release Branches
 
-### Merging Release Branch Into Develop
+To not have to merge bug fixes into several branches and create several pull requests, release branch should be merged
```
... **the** release branch ...?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+        git pull <remote>/r/6.x
 
-2. Switch to commit you want to create the release from:
+2. Add the release notes, and update the changelog. The `create-changelog` [helper script
```
What about adding a step to merge down release branches (e.g. `r/5.x` into `r/6.x`) before the release is tagged?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-If the version is a release candidate, you probably want to highlight that there are no *Blockers* left at the moment
-and *#propose* this to become the final release.
+Updating the [localization translations](localization.md) is automated for existing translation files. If new files need
+to be added, it is something that should happen early during the release process.
```
Deletion of translations is also a manual procedure.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
-        git checkout 060dfc3e2328518ae402846577fc6e7ce3b650d7
+        vim docs/guides/admin/docs/releasenotes.md
+        vim docs/guides/admin/docs/changelog.md
+        git commit -S docs/guides/admin/docs/releasenotes.md docs/guides/admin/docs/changelog.md
```
Maybe also add a message?

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-12. Push the built artifacts to Maven. Bug the QA Coordinator to do this so that he remembers to set this up from the CI
-    servers.
+9. Push the tag to the community repository (you can remove the branch afterwards):
```
`7.`

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
 
-14. Update [the website](https://github.com/opencast/opencast.github.io) with a pull request, or directly if you have
-    the appropriate permissions.
+10. Create a new release on Github using the [graphical user interface](https://github.com/opencast/opencast/releases)
```
`8.`

mtneug replied:
Git**H**ub

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,366 @@
+Subject: Opencast <VERSION> Released
 
-    The documentation for this release can be found at [http://docs.opencast.org].
+Hi all,
```
I guess it doesn't matter, but `Hi everyone` is used elsewhere.

mtneug reviewed (dismissed):
In general, I think this is a good description of the release manager work.

lkiesow committed:
MH-13307, More Release Documentation Fixes

This patch adds a few minor fixes to the release manager documentation.

mtneug commented on docs/guides/developer/docs/release-manager.md:
```diff
@@ -1,474 +1,377 @@
-    Subject: Opencast <VERSION>: Language translation status
 
-    Hi everyone,
+### Status of Translations
```
What about regular updates about the translation status? The release manager should also point the various ways to contribute to the translations:

* translate strings
* verify translations
* help clear up issues of fellow translators

lkiesow replied:
I think that's a good idea but where would you put this information? Right now, we are sending out several emails. Would you suggest to include it in all of the emails? Or send out an additional mail? Or send all information (languages endangered, languages included and how to contribute) in a single email instead?

mtneug replied:
As was done in the past, I regularly sent reminder status update mails ([1](https://groups.google.com/a/opencast.org/forum/?fromgroups=#!topic/users/bH8vpkaVMJo), [2](https://groups.google.com/a/opencast.org/forum/?fromgroups=#!topic/users/VwOxqB-8l_o), [3](https://groups.google.com/a/opencast.org/forum/?fromgroups=#!topic/users/mSPhAFxT8fc), [4](https://groups.google.com/a/opencast.org/forum/?fromgroups=#!topic/users/ynhbT2BmNdM)) so that would be an option. The [translation week mail](https://groups.google.com/a/opencast.org/forum/?fromgroups=#!topic/users/Kjp2WB4GAI0) is another one. In this example, only the "help clear up issues of fellow translators" is missing.

lkiesow committed:
MH-13307, More Release Documentation Fixes

This patch adds a few minor fixes to the release manager documentation.

lkiesow committed:
MH-13307, Document Translation Announcements

This patch adds an additional email template for the translation week.

lkiesow commented:
The website now is the last open question. Let's wait until Thursday and see how the discussion on list evolves.

mtneug commented:
@lkiesow have you seen some of my other comments?

Also, what about JAR upload to Nexus? I'm still planning to automate this step so I guess we could leave it out?

lkiesow commented:
> @lkiesow have you seen some of my other comments?

Which of your comments did I not address? I thought I got all of them.

> Also, what about JAR upload to Nexus? I'm still planning to automate this step so I guess we could leave it out?

Definitely. Most people do not even have the permissions to do that.

mtneug commented:
> Which of your comments did I not address? I thought I got all of them.

There are still some unresolved comments from Greg and me ([1](https://github.com/opencast/opencast/pull/648/files#r247620643), [2](https://github.com/opencast/opencast/pull/648/files#r247621208), [3](https://github.com/opencast/opencast/pull/648/files#r249293414), [4](https://github.com/opencast/opencast/pull/648/files#r249293421), [5](https://github.com/opencast/opencast/pull/648/files#r249293428), [6](https://github.com/opencast/opencast/pull/648/files#r249293473), [7](https://github.com/opencast/opencast/pull/648/files#r249293622), [8](https://github.com/opencast/opencast/pull/648/files#r249293591)).

lkiesow committed:
MH-13307, Even More Release Manager Documentation Fixes

This patch fixes some additional review findings.

lkiesow commented:
Ups, I ignored the ones hidden by Github. Sorry.
I've now addressed those as well.
